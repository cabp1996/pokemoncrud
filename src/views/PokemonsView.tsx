import React from "react";
import { useSelector } from "react-redux";
import { PokemonTable } from "../components";
import { PokemonForm } from "../components";
import { Header } from "../components";
import { RootState } from "../reducers/root.reducer";

export const PokemonsView = () => {
  const { showPokemonForm } = useSelector((state: RootState) => state.pokemons);

  return (
    <div className="main-container">
      <Header />
      <PokemonTable />
      {showPokemonForm && <PokemonForm />}
    </div>
  );
};
