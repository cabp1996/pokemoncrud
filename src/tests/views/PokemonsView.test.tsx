import { render, RenderResult } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { PokemonsView } from "../../views";
import { initialState, pokemonsMockData } from "../mocks/initial-state.mock";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let store = mockStore(initialState);
store.dispatch = jest.fn();

describe("PokemonsView Tests", () => {
  let wrapper: RenderResult;

  beforeEach(async () => {
    window.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve(pokemonsMockData),
      })
    ) as jest.Mock;
  });

  beforeEach(() => {
    wrapper = render(
      <Provider store={store}>
        <PokemonsView />
      </Provider>
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
