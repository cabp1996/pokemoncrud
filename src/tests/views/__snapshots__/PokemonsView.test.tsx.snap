// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonsView Tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-container"
      >
        <h1>
          Listado de Pokemon
        </h1>
        <div
          class="search-bar-container"
        >
          <div
            class="search-input"
          >
            <input
              class="form-input"
              name="filter"
              placeholder="Buscar"
              type="text"
              value=""
            />
          </div>
          <button
            class="main-button"
          >
            <i
              aria-hidden="true"
              class="fa fa-plus"
            />
             Nuevo
          </button>
        </div>
        <table
          class="pokemon-table"
        >
          <thead>
            <tr>
              <th>
                Nombre
              </th>
              <th>
                Imagen
              </th>
              <th>
                Ataque
              </th>
              <th>
                Defensa
              </th>
              <th>
                Acciones
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                name1
              </td>
              <td>
                <img
                  alt="name1"
                  src="image1"
                />
              </td>
              <td>
                50
              </td>
              <td>
                50
              </td>
              <td>
                <div
                  class="td-actions"
                >
                  <button
                    class="btn-icon"
                  >
                    <i
                      aria-hidden="true"
                      class="fa fa-pencil"
                    />
                  </button>
                  <button
                    class="btn-icon"
                  >
                    <i
                      aria-hidden="true"
                      class="fa fa-trash"
                    />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                name2
              </td>
              <td>
                <img
                  alt="name2"
                  src="/images/notfound.png"
                />
              </td>
              <td>
                50
              </td>
              <td>
                50
              </td>
              <td>
                <div
                  class="td-actions"
                >
                  <button
                    class="btn-icon"
                  >
                    <i
                      aria-hidden="true"
                      class="fa fa-pencil"
                    />
                  </button>
                  <button
                    class="btn-icon"
                  >
                    <i
                      aria-hidden="true"
                      class="fa fa-trash"
                    />
                  </button>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-container"
    >
      <h1>
        Listado de Pokemon
      </h1>
      <div
        class="search-bar-container"
      >
        <div
          class="search-input"
        >
          <input
            class="form-input"
            name="filter"
            placeholder="Buscar"
            type="text"
            value=""
          />
        </div>
        <button
          class="main-button"
        >
          <i
            aria-hidden="true"
            class="fa fa-plus"
          />
           Nuevo
        </button>
      </div>
      <table
        class="pokemon-table"
      >
        <thead>
          <tr>
            <th>
              Nombre
            </th>
            <th>
              Imagen
            </th>
            <th>
              Ataque
            </th>
            <th>
              Defensa
            </th>
            <th>
              Acciones
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              name1
            </td>
            <td>
              <img
                alt="name1"
                src="image1"
              />
            </td>
            <td>
              50
            </td>
            <td>
              50
            </td>
            <td>
              <div
                class="td-actions"
              >
                <button
                  class="btn-icon"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-pencil"
                  />
                </button>
                <button
                  class="btn-icon"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-trash"
                  />
                </button>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              name2
            </td>
            <td>
              <img
                alt="name2"
                src="/images/notfound.png"
              />
            </td>
            <td>
              50
            </td>
            <td>
              50
            </td>
            <td>
              <div
                class="td-actions"
              >
                <button
                  class="btn-icon"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-pencil"
                  />
                </button>
                <button
                  class="btn-icon"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-trash"
                  />
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
